docs/build-instructions.md

```diff
@@ -2,126 +2,202 @@
 
+
 ## Quick start with Docker
 
 * Install SGX SDK
 
 ```shell
-$ wget https://download.01.org/intel-sgx/sgx-dcap/1.12.1/linux/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.15.101.1.bin
+wget https://download.01.org/intel-sgx/sgx-dcap/1.12.1/linux/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.15.101.1.bin
 
-# choose to install the sdk into the /opt/intel
-$ chmod a+x ./sgx_linux_x64_sdk_2.15.101.1.bin && sudo ./sgx_linux_x64_sdk_2.15.101.1.bin
+#choose to install the sdk into the /opt/intel
+chmod a+x ./sgx_linux_x64_sdk_2.15.101.1.bin && sudo ./sgx_linux_x64_sdk_2.15.101.1.bin
 
-$ source /opt/intel/sgxsdk/environment
+source /opt/intel/sgxsdk/environment
 ```
 
 * Build the ehsm-kms docker image
 ```shell
-$ git clone https://github.com/intel/ehsm.git ehsm && cd ehsm
+git clone https://github.com/intel/ehsm.git ehsm && cd ehsm
+
+./docker/core/build_and_run.sh -b
+#it will generate the docker image <ehsm_kms_service.tar.gz> under ./docker/core
+```
+
+* Install CouchDB
+```shell
+docker pull couchdb
 
-$ ./docker/build_and_run.sh -b
+#change the user, password, and port here if needed
+docker run --name my-couchdb -e COUCHDB_USER=admin -e COUCHDB_PASSWORD=password -p 5984:5984 -d couchdb
+```
 
-it will generate the docker image <ehsm_kms_service.tar.gz> under your workdir.
+* Change corresponding user, passwork, IP, etc. in build_and_run.sh
+``` shell
+vim docker/core/build_and_run.sh
+```
+``` vim
+EHSM_CONFIG_COUCHDB_USERNAME="admin" (your username)
+EHSM_CONFIG_COUCHDB_PASSWORD="password" (your password)
+EHSM_CONFIG_COUCHDB_SERVER="1.2.3.4" (your IP)
+EHSM_CONFIG_COUCHDB_PORT="5984" (your port)
 ```
+
 * Run the ehsm-kms service with docker
 ```shell
-$ ./docker/build_and_run.sh -r 
+cd ./docker/core
+./build_and_run.sh -r 
 or
-$ ./docker/build_and_run.sh -r ehsm_kms_service.tar.gz
+./build_and_run.sh -r ehsm_kms_service.tar.gz
 
-$ docker ps
-CONTAINER ID   IMAGE                     COMMAND                  CREATED         STATUS         PORTS                                       NAMES
-6b0f786713e5   ehsm_kms_service:latest   "node ehsm_kms_serve…"   8 seconds ago   Up 6 seconds   0.0.0.0:9000->9000/tcp, :::9000->9000/tcp   tender_spence
+docker ps
+CONTAINER ID   IMAGE                     COMMAND                  CREATED         STATUS         PORTS                                                               NAMES
+035bd3b208bb   ehsm_kms_service:latest   "sh /home/start_ehsm…"   1 minutes ago    Up 1 minutes    0.0.0.0:9000->9000/tcp, :::9000->9000/tcp                             heuristic_germain
+72b9f592eff7   couchdb                   "tini -- /docker-ent…"   10 minutes ago   Up 10 minutes   4369/tcp, 9100/tcp, 0.0.0.0:5984->5984/tcp, :::5984->5984/tcp   my-couchdb
 ```
 
-* Run the unit-test cases (you can do it in another remote device)
-```
-$ cd ehsm_kms_service/test
-$ python3 test_kms_with_rest.py -i <your-kms-docker-ip> -p <your-kms-port>
+* Wait about 1 min and check whether the eHSM is listening
+``` shell
```

Comment:
hanyu, for the single machine, you need to run the dkeyserver and dkeycache.

if you following that way, it don't need to wait 1min for kms docker.